ui/apps/profile/src/components/pages/profile-info-page.tsx

```diff
@@ -86,14 +108,69 @@ const ProfileInfoPage: React.FC<RootComponentProps> = props => {
   const background = getProfileImageVersionsWithMediaUrl(profileData?.background);
   const avatar = getProfileImageVersionsWithMediaUrl(profileData?.avatar);
 
-  return (
-    <>
+  if (profileDataReq.data && profileData)
+    return (
+      <>
+        <Stack direction="column" spacing="gap-y-4" fullWidth>
+          <ProfileHeader
+            did={profileData.did}
+            background={background}
+            avatar={avatar}
+            name={profileData.name}
+            ensName={null /*@TODO: integrate ENS when the API is ready */}
+            viewerIsOwner={isViewer}
+            flagLabel={t('Report')}
+            copyLabel={t('Copy to clipboard')}
+            copiedLabel={t('Copied')}
+            followElement={
+              <FollowProfile profileId={profileId} isIconButton={true} navigateTo={navigateTo} />
+            }
+            handleFlag={checkAuth(
+              handleEntryFlag(
+                profileData.did.id ? profileData.did.id : '',
+                EntityTypes.PROFILE,
+                profileData.name,
+              ),
+            )}
+            handleEdit={() => {
+              navigateTo({
+                appName: '@akashaorg/app-profile',
+                getNavigationUrl: () => `/${profileId}${routes[EDIT]}`,
+              });
+            }}
+          />
+          {profileData.description && (
+            <ProfileBio title={t('Bio')} biography={profileData.description} />
+          )}
+          <ProfileStatsPresentation profileId={profileId} navigateTo={navigateTo} />
+          {profileData.links?.length > 0 && (
+            <ProfileLinks
+              title={t('Find me on')}
+              links={profileData.links}
+              copiedLabel={t('Copied')}
+              copyLabel={t('Copy to clipboard')}
+            />
+          )}
+          {showFeedback && (
+            <Snackbar
+              title={t('You unfollowed {{name}}', { userName: profileData.name })}
+              iconType="CheckCircleIcon"
+              handleDismiss={() => {
+                setShowFeedback(false);
+              }}
+              customStyle="mb-4"
+            />
+          )}
+        </Stack>
+      </>
+    );
+
+  if (validDid) {
+    return (
```

Comment:
This can be simplified as follows:

First, declare the following variables: 

`const hasProfile = status === 'success' && profileData;`

`const profileNotCreated = (!isLoggedIn && !validDid && !hasProfile) || (isLoggedIn  && !hasProfile);` You can use this variable to conditionally display `DefaultEmptyCard` component.

`const profile = profileNotCreated ? { did : { id: profileId } } : profileData`. Then you can use this with optional chaining on `ProfileHeader` component. 

That will make the code cleaner and easier to read.